clr/commands.py

```diff
@@ -326,20 +339,22 @@ def cmd_argtest(self, a, b, c=4, d=None, e=False)
                         f"Unexpected kwarg **{name} in {command_name}."
                     )
 
-                # Assume string type for most args.
-                default_type = str
-                if param.default is not None:
-                    default_type = type(param.default)
-                if default_type not in (str, bool, int, float):
-                    raise AssertionError(
-                        f"Unexpected arg type for {name} in {command_name}: "
-                        f"{default_type}"
-                    )
+                if not arg_type and param.default is not None:
+                    # Infer from type of default
+                    arg_type = type(param.default)
```

Comment:
Maybe this + L289-L299 can be consolidated into a single helper function like `_get_arg_type` so that all argument type handling can be in one place

Reply:
Good idea! Done